tests/test-filo.c

```diff
@@ -0,0 +1,50 @@
+#include "test.h"
+
+#include <utils/filo.h>
+#include <utils/utils.h>
+
+#include <string.h>
+
+#define NUM_TESTS               100
+
+static int test_push(filo_t *pfilo, int elem)
+{
+    return filo_push(pfilo, &elem);
+}
+
+static int test_pop(filo_t *pfilo, int *pelem)
+{
+	return filo_pop(pfilo, pelem, true);
+}
+
+void do_test_filo(test_t *ctx, test_result_t *result)
+{
+	filo_t *pfilo = filo_alloc(sizeof(int), NUM_TESTS);
+    int i;
+	int total=0, pass=0;
+
+    for (i = 0; i < NUM_TESTS; i++) {
+        if (test_push(pfilo, i + 1) == 0)
```

Comment:
There are some macros to help with these result updates and reporting in `test.h` if those macro's APIs are restrictive, please feel free to add as needed.